modules/uhttp/http.go

```diff
@@ -160,17 +158,15 @@ func (m *Module) Initialize(host service.Host) error {
 func (m *Module) Start(ready chan<- struct{}) <-chan error {
 	m.mux = http.NewServeMux()
 	// Do something unrelated to annotations
-	m.router = mux.NewRouter()
+	m.router = NewRouter(m.Host())
 
 	m.mux.Handle("/", m.router)
 
 	for _, h := range m.handlers {
 		handle := newExecutionChain(m.filters, h.Handler)
-		m.router.Handle(h.Path, handle)
+		m.router.HandleRoute(m.Host(), h.Path, handle)
```

Comment:
Why do we have to pass the host as an argument when the router already has a reference to the host?